olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -843,6 +851,16 @@ DefaultCache::StorageOpenResult DefaultCacheImpl::SetupProtectedCache() {
     return ToStorageOpenResult(status);
   }
 
+  if (!protected_cache_->CheckState()) {
+    OLP_SDK_LOG_ERROR_F(kLogTag,
+                        "Check state for protected cache failed there are more "
+                        "than 4 L0 files present in %s",
+                        settings_.disk_path_protected.get().c_str());
+
+    protected_cache_.reset();
+    return StorageOpenResult::ProtectedCacheCorrupted;
+  }
+
```

Comment:
Please revert #1217 and just update CheckCompactionFinshed method to allow up to 4 files on L0